draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -385,128 +355,160 @@ plus a string suitable for computing a digest, but such approaches can easily le
+
+For decoy digests, no Disclosure is sent to the Holder, i.e., the Holder will see hash digests that do not correspond to any Disclosure. See (#decoy_digests_privacy) for additional privacy considerations.
+
+To ensure readability and replicability, the examples in this specification do not contain decoy digests unless explicitly stated.
+
+#### Creating an SD-JWT {#creating_sd_jwt}
+
+An SD-JWT is a JWT that MUST be signed using the Issuer's private key.
+
+An SD-JWT MAY contain both selectively disclosable claims and non-selectively disclosable claims, i.e., claims that are always contained in the SD-JWT in plaintext and are always visible to a Verifier.
+
+Claims controlling the validity of the SD-JWT, such as `iss`, `exp`, or `nbf` are usually included in plaintext. End-User claims MAY be included as plaintext as well, e.g., if hiding the particular claims from the Verifier does not make sense in the intended use case. Ultimately, an Issuer decides which claims are selectively disclosable and which are not.
+
+Plaintext claims are included in the SD-JWT just as they would be in any other JWT.
+
+Selectively disclosable claims are omitted from the SD-JWT. Instead, the hash digests of the respective Disclosures and potentially decoy digests are contained as an array in a new claim, `_sd`.
+
+The `_sd` claim MUST be an array of strings, each string being a hash digests of a Disclosure or a decoy digest as described above.
+The array MAY be empty, although it is RECOMMENDED to omit the claim in this case to save space.
+
+The Issuer MUST hide the original order of the claims in the array. To ensure this, it is RECOMMENDED to shuffle the array, e.g., by sorting it alphanumerically or randomly. The precise method does not matter as long as it does not depend on the original order of elements.
+
+Issuers MUST NOT issue SD-JWTs where
+
+ * the key `_sd` is used for any other purpose than to contain the array of hash digests, or
+ * the key `_sd` is itself used inside a claim value (when the claim value is an object), or
+ * the same Disclosure appears more than once (in the same array or in different arrays).
+
+
+#### Nested Data in SD-JWTs {#nested_data}
+
+Just like any JWT, an SD-JWT MAY contain objects that themselves contain other data. For any object in an SD-JWT, the Issuer MAY decide to either make the entire object selectively disclosable or to make its properties selectively disclosable individually. In the latter case, the Issuer MAY even choose to make some some of the object's properties selectively disclosable and others not.
```

Comment:
`objects that themselves contain other data` -> something like `key value pairs where the value is an object` or `where the claim value is expressed as an object` might be clearer, no?

Reply:
Agree.

Reply:
```suggestion
Just like any JWT, an SD-JWT MAY contain key value pairs where the value is an object. For any object in an SD-JWT, the Issuer MAY decide to either make the entire object selectively disclosable or to make its properties selectively disclosable individually. In the latter case, the Issuer MAY even choose to make some some of the object's properties selectively disclosable and others not.
```